horovod/common/operations.cc

```diff
@@ -235,7 +235,7 @@ bool IncrementTensorCount(std::unique_ptr<MessageTable>& message_table,
 
   std::vector<Request>& messages = std::get<0>(table_iter->second);
   int count = (int)messages.size();
-  bool ready_to_reduce = count == mpi_size;
+  bool ready_to_reduce = count == (mpi_size - joined_size);
```

Comment:
What happens with allgather & broadcast?  Could we support them as well?

Reply:
I haven't added support for allgather & broadcast yet, but it should be relatively easy after allreduce.
I hope to get the allreduce PR in and then add support for allgather & broadcast & non-PyTorch frameworks in subsequent PRs.

Reply:
Hello, please, if you want to support tensorflow, is it necessary to implement a join op.
E.g:
Class HorovodJoinOp : public AsyncOpKernel{}

Reply:
@huoliquankai7 In the end TensorFlow will be supported, this particular PR supports only PyTorch.